code/game/objects/structures/beds_chairs/chair.dm

```diff
@@ -112,6 +113,11 @@
 	else
 		rotate()
 
+/obj/structure/chair/relaymove(mob/living/user)
```

Comment:
Don't duplicate the code, either make it on obj/structure with a variable saying it should activate on chairs and beds or make it a component (better)

Reply:
COMPONENTTTTTt